.pre-commit-config.yaml

```diff
@@ -1,7 +1,55 @@
-fail_fast: true
 repos:
-- hooks:
-  - id: black
-    language_version: python3
-  repo: https://github.com/psf/black
-  rev: 20.8b1
+  - repo: local
+    hooks:
+      - id: static_word_checks
+        args:
+          - '--replace'
+        entry: python -m tools.static_word_checks
+        language: system
+        name: Static Word Checks
+        pass_filenames: false
+        types: [ python ]
+
+      - id: check_documentation
+        entry: python -m tools.check_documentation
+        language: system
+        name: Check Documentation
+        pass_filenames: false
+        types: [ python ]
+
+  - repo: https://github.com/psf/black
+    hooks:
+      - id: black
+        language_version: python3
+    rev: 21.7b0
```

Comment:
is it possible to make this work without setting the value `rev`?
It would add technical debt..

Reply:
It's set by `pre-commit autoupdate`.

Reply:
Requiring hooks to have pinned versions is a core design decision of pre-commit, since it ensures that all developers and CIs get the same consistent version of hooks for a given project repository commit, and thus hooks don't arbitrarily fail or make different changes based on whatever version happens to be installed on the current platform. There's also a GH action that, like dependabot, automatically submits PRs at a reasonable frequency to upgrade out of date hooks.